### PR TITLE
Deduplicate command names in generated LSP support doc

### PR DESCRIPTION
**Pull request by timothyhutz**
Using unique from itertools::Itertools function to clear out and duplicate string entries. Tested on my ARM64 Windows 11 system with mdbook and generated via xtask docgen.

### CODE REVIEW

**Comment by timothyhutz** [comment]
This is my contributions to the project for Issue https://github.com/helix-editor/helix/issues/10316

File: xtask/Cargo.toml
Comment (pascalkuthe) [inline xtask/Cargo.toml:20]: we don't want to pull in extra dependencies, particularly not itertools

Comment (timothyhutz) [reply xtask/Cargo.toml:20]: ok so use what's available to remove the Vector Duplication string?

Comment (kirawi) [reply xtask/Cargo.toml:20]: I'm not on my computer right now, but if that's a vector then it already has `dedup` methods

Comment (kirawi) [reply xtask/Cargo.toml:20]: You could possibly collect to a `HashMap` or a `IndexMap` (if order is important)?

Comment (timothyhutz) [reply xtask/Cargo.toml]: OK redid it without itertools. let me know what you think.

File: xtask/src/docgen.rs
Comment (kirawi) [inline xtask/src/docgen.rs:101]:
```suggestion
                            .collect::<HashSet<_>>();
```
Is what I meant

Comment (kirawi) [reply xtask/src/docgen.rs:101]: Hmm, actually substituting the `join` functionality is a pain until `intersperse lands`

Comment (timothyhutz) [reply xtask/src/docgen.rs:101]: Yep, had to convert to slice first to join them as a string.. Does Rust have a stringify method to use? might be simpler way to convert list to comma string.

Comment (timothyhutz) [reply xtask/src/docgen.rs:101]: If your happy with it like this can we merge it?

File: xtask/Cargo.toml
Comment (the-mikedavis) [inline xtask/Cargo.toml:1]: This change should be dropped

File: xtask/src/docgen.rs
Comment (the-mikedavis) [inline xtask/src/docgen.rs:4]: Same here, this change can be reset

File: xtask/src/docgen.rs
Comment (the-mikedavis) [inline xtask/src/docgen.rs:102]: I would prefer that we keep the order that we have in the `languages.toml` file for this generated doc since definition order is somewhat important (for example code actions and symbol pickers are ordered by them).

We can use a hash set that we insert names into as we go so that we keep the order and remove duplicates:

```rust
let mut seen_commands = HashSet::new();
let mut commands = String::new();
for ls_config in lc
    .language_servers
    .iter()
    .filter_map(|ls| config.language_server.get(&ls.name))
{
    let command = &ls_config.command;
    if !seen_commands.insert(command) {
        continue;
    }

    if !commands.is_empty() {
        commands.push_str(", ");
    }

    commands.push_str(&md_mono(command));
}
row.push(commands);
```

Comment (timothyhutz) [reply xtask/src/docgen.rs:102]: This is my first RUST Foss change contribution.. If you want me to redo it again to use HashMap instead I can.. I was looking for the most simple approach.

File: xtask/Cargo.toml
Comment (timothyhutz) [reply xtask/Cargo.toml:1]: Why?

File: xtask/src/docgen.rs
Comment (RoloEdits) [reply xtask/src/docgen.rs:102]: You should be able to copy that code in and it should do the job. The `HashSet` code.

File: xtask/Cargo.toml
Comment (RoloEdits) [reply xtask/Cargo.toml:1]: Its not changing anything in the file, nor is it contributing to the change in the pr, therefore its best to remove the file from the commit/s. This only dirties histories of files.

Comment (timothyhutz) [reply xtask/Cargo.toml:1]: I rebased and fixup'd everything into 1 commit which drops the unneeded commits.

File: xtask/src/docgen.rs
Comment (timothyhutz) [reply xtask/src/docgen.rs]: ok I implemented your code using hashset. xtask docgen worked..

**Comment by timothyhutz** [comment]
Ready for second peer review please.

File: xtask/Cargo.toml
Comment (kirawi) [inline xtask/Cargo.toml:19]: Can you undo this change?

File: xtask/Cargo.toml
Comment (the-mikedavis) [reply xtask/Cargo.toml:1]: It seems like those commits might have been squashed into the single commit instead - those whitespace changes are still in the diff

File: xtask/Cargo.toml
Comment (timothyhutz) [reply xtask/Cargo.toml]: amended commit pushed

**Comment by timothyhutz** [comment]
ready for another PR review.

**Review by the-mikedavis — APPROVED** [review]

**Comment by timothyhutz** [comment]
can I get another approver pls?

**Pull request merged**
